src/fidesops/service/connectors/saas_connector.py

```diff
@@ -98,40 +104,55 @@ def create_client(self) -> AuthenticatedClient:
-        """Helper method to extract an arbitrary data path from a given dictionary"""
-        value = dictionary
-        for key in path.split("/"):
-            value = value[key]
-        return value
-
-    def retrieve_data(
-        self, node: TraversalNode, policy: Policy, input_data: Dict[str, List[Any]]
+    def retrieve_data(  # pylint: disable=R0914
+        self,
+        node: TraversalNode,
+        policy: Policy,
+        request: PrivacyRequest,
+        input_data: Dict[str, List[Any]],
     ) -> List[Row]:
         """Retrieve data from SaaS APIs"""
 
         # get the corresponding read request for the given collection
-        collection_name = node.address.collection
-        read_request = self.endpoints[collection_name].requests["read"]
+        collection_name: str = node.address.collection
+        read_request: SaaSRequest = self.endpoints[collection_name].requests["read"]
 
-        query_config = self.query_config(node)
+        query_config: SaaSQueryConfig = self.query_config(node)
         prepared_requests = query_config.generate_query(input_data, policy)
 
         rows: List[Row] = []
         for prepared_request in prepared_requests:
             response = self.client().get(prepared_request)
 
-            # process response
-            if read_request.data_path:
-                processed_response = self.get_value_by_path(
-                    response.json(), read_request.data_path
+            if read_request.postprocessors is None:
+                rows.extend(response.json())
+                continue
+            data_to_be_processed = response.json()
+            length_postprocessors = len(read_request.postprocessors)
+            for post_processor in read_request.postprocessors:
+                strategy: PostProcessorStrategy = get_strategy(
+                    post_processor.strategy, post_processor.configuration
                 )
-            else:
-                # by default, we expect the collection_name to be one of the root fields in the response
-                processed_response = response.json()[collection_name]
-
-            rows.extend(processed_response)
+                logger.info(
+                    f"Starting postprocessing with strategy {strategy.get_strategy_name()}"
+                )
+                try:
+                    processed_response = strategy.process(
+                        data_to_be_processed, request.get_cached_identity_data()
+                    )
+                    if processed_response:
+                        # if last postprocessor
+                        if (
+                            read_request.postprocessors.index(post_processor) + 1
+                            == length_postprocessors
+                        ):
+                            rows.extend(processed_response)
+                        else:
+                            data_to_be_processed = processed_response
+                except Exception as e:
```

Comment:
The order will be hard-coded in the config, so for example, here, unwrap then filter will be run:

```
"postprocessors": [
                   {
                       "strategy": "unwrap",
                        "configuration": {
                            "data_path": "conversation_messages"
                        }
                   },
                   {
                       "strategy": "filter",
                       "configuration": {
                           "field": "from_email",
                           "value": {
                               "identity": "email"
                           }
                       }
                   }
               ]
```

What do you mean by unintentional effects?

Reply:
OK thanks for showing me how the order is specified.  

To rephrase, `unwrap` *has* to be run first in this example.  Your filter method only has the ability to filter on top level data, and the unwrap method is specified to only work on dictionary data, the postprocessors aren't interchangeable on the data it can work with.  I guess good documentation around how to specify the order of your postprocessors would help here, so that users are aware that the order matters.

Reply:
Yes definitely. I'll write up some docs today